remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 Parsley Debug View Contributors <https://github.com/j-mie6/parsley-debug-views/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debug
+
+import parsley.debug.internal.DebugTreeSerialiser
+
+import sttp.client3.*
+
+import scala.util.Try
+import scala.concurrent.duration.*
+
+/** The RemoteView HTTP module allows the parsley debug tree to be passed off to a server through a specified port on
+  * local host (by default) or to a specified IP address. This enables all of the debug tree parsing, serving and 
+  * graphics to be done in a separate process, and served reactively to a client.
+  * 
+  * This module is part of the Debugging Interactively in parsLey Library (DILL) project,
+  * (https://github.com/j-mie6/parsley-debug-app).
+  * 
+  * RemoteView uses the STTP library to create HTTP requests to a specified IP address, over a specified port.
+  * The request is formatted using the upickle JSON formatting library, it is being used over other
+  * libraries like circe for its improved speed over large data structures.
+  */
+sealed trait RemoteView extends DebugView.Reusable {
+  implicit protected val port: Integer
+  implicit protected val address: String
+
+  /**
+    * Send the debug tree and input to the port and address specified in the 
+    * object construction. 
+    *
+    * @param input The input source.
+    * @param tree The debug tree.
+    */
+  override private [debug] def render(input: => String, tree: => DebugTree): Unit = {
+    // Printing helpers
+    val TEXT_TO_RED: String    = "\u001b[31m"
+    val TEXT_TO_NORMAL: String = "\u001b[0m"
+
+    val CONNECTION_TIMEOUT: FiniteDuration = 30.second
+    val RESPONSE_TIMEOUT: FiniteDuration = 10.second
+
+    // Create localhost endpoint using port
+    val sb: StringBuilder = new StringBuilder("http://")
+    sb.append(address)
+    sb.append(':')
+    sb.append(port.toString())
+    sb.append("/remote")
+    val endPoint: String = sb.result()
+
+    // Format payload as JSON
+    sb.clear()
+    sb.append("{\"input\": \"")
+    sb.append(input)
+    sb.append("\", \"tree\": ")
+    sb.append(DebugTreeSerialiser.toJSON(tree))
+    sb.append("}")
+    val payload: String = sb.result()
+
+    println("Sending Debug Tree to Server")
+
+    // Send POST
+    try {
+      val backend = HttpURLConnectionBackend(
```

Comment:
```suggestion
      val backend = TryBackend(HttpURLConnectionBackend(
```

This would avoid the need for the `try` above, and allows for the more idiomatic `Try` combinators.

Reply:
👍